src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -16,61 +16,63 @@ public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : Diagnost
+                                                                                          s_localizableDoNotUseWeakAlgorithmsDescription);
 
         internal static DiagnosticDescriptor DoNotUseRIPEMD160SpecificRule = CreateDiagnosticDescriptor(DoNotUseWeakCryptographicRuleId,
-                                                                                                s_localizableDoNotUseRIPEMD160Title,
-                                                                                                s_localizableDoNotUseRIPEMD160Description);
+                                                                                          s_localizableDoNotUseWeakAlgorithmsTitle,
+                                                                                          s_localizableDoNotUseWeakAlgorithmsMessage,
+                                                                                          s_localizableDoNotUseWeakAlgorithmsDescription);
 
         internal static DiagnosticDescriptor DoNotUseDSASpecificRule = CreateDiagnosticDescriptor(DoNotUseBrokenCryptographicRuleId,
-                                                                                          s_localizableDoNotUseDSATitle,
-                                                                                          s_localizableDoNotUseDSADescription);
+                                                                                          s_localizableDoNotUseBrokenAlgorithmsTitle,
+                                                                                          s_localizableDoNotUseBrokenAlgorithmsMessage,
+                                                                                          s_localizableDoNotUseBrokenAlgorithmsDescription);
 
         protected abstract Analyzer GetAnalyzer(CompilationStartAnalysisContext context, CompilationSecurityTypes cryptTypes);
 
         private static readonly ImmutableArray<DiagnosticDescriptor> s_supportedDiagnostics = ImmutableArray.Create(DoNotUseMD5SpecificRule,
-                                                                                                                  DoNotUseDESSpecificRule,
-                                                                                                                  DoNotUseRC2SpecificRule,
-                                                                                                                  DoNotUseTripleDESSpecificRule, 
-                                                                                                                  DoNotUseRIPEMD160SpecificRule,
-                                                                                                                  DoNotUseDSASpecificRule);
+                                                                                                                    DoNotUseDESSpecificRule,
```

Comment:
This is wrong - these aren't five separate rules -  it's one rule (defined by it's one id). We should have just one descriptors that's returned from here.